app/services/state.py

```diff
@@ -35,6 +35,34 @@ async def delete_state(self, integration_id: str, action_id: str, source_id: str
                     f"integration_state.{integration_id}.{action_id}.{source_id}"
                 )
 
+    async def group_add(self, group_name: str, values: list):
```

Comment:
This looks reasonable. Do you imagine this going into the template repository?

Reply:
Yes. The plan is to try it in the ATS integration, get some feedback, make improvements as needed, and then move it to the template repo so anyone can use it in similar scenarios.